### PR TITLE
[front] - feature(components/assistant_builder): enhance text editor with shortcuts

### PR DESCRIPTION
**Pull request by JulesBelveze**
## Description

 - Introduce `History` extension to enable undo/redo features in the text editor
 - Extend the `Paragraph` extension to include a keyboard shortcut for inserting a line break with 'Shift-Enter'
 - Refacto: extract custom extensions for reusability

## Deploy Plan

Deploy `front`

### CODE REVIEW

File: front/components/assistant_builder/InstructionScreen.tsx [hunk 1]
```diff
@@ -127,7 +129,19 @@ export function InstructionScreen({
   instructionsError: string | null;
 }) {
   const editor = useEditor({
-    extensions: [Document, Text, Paragraph],
+    extensions: [
+      Document,
+      Text,
+      Paragraph,
+      History,
+      HardBreak.extend({
```
Comment (flvndvd) [inline front/components/assistant_builder/InstructionScreen.tsx:137]: Let's extract those extensions so we can re-use them in different places.

Comment (JulesBelveze) [reply front/components/assistant_builder/InstructionScreen.tsx]: @flvndvd how about now?

**Comment by spolu** [comment]
deferring review to @flvndvd

File: front/components/wysiwyg/editor.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+import { History } from "@tiptap/extension-history";
+import Paragraph from "@tiptap/extension-paragraph";
+import Text from "@tiptap/extension-text";
+
+export const CONTENT_EDITOR_EXTENSIONS = [
+  Document,
+  Text,
+  Paragraph,
+  History,
+  HardBreak.extend({
```
Comment (flvndvd) [inline front/components/wysiwyg/editor.ts:12]: No need for hardbreak, you can override shortcuts for `Paragraph`. Also adding `Hardbreak` would have required to update how we convert Tip Tap nodes to raw text.

**Review by flvndvd — APPROVED** [review]
LGTM, please update the PR description.

**Pull request merged**